Postgres-based internal API servers will not start due to client mismatch

**Describe the bug**
Duchy servers which use Google Cloud Storage will not start.

The assumption is that a transitive Maven dep was updated, as the project does not currently pin Maven deps.

**Component(s) affected**
Duchy

**Version**
HEAD (d4b9a656b2d9dfab384de5130d9643c3e3bccd03)

**Environment**
halo-cmm-dev on GKE

**Additional context**
```
Exception in thread "main" java.lang.ExceptionInInitializerError
	at com.google.api.services.storage.Storage$Builder.build(Storage.java:10534)
	at com.google.cloud.storage.spi.v1.HttpStorageRpc.<init>(HttpStorageRpc.java:127)
	at com.google.cloud.storage.StorageOptions$DefaultStorageRpcFactory.create(StorageOptions.java:61)
	at com.google.cloud.storage.StorageOptions$DefaultStorageRpcFactory.create(StorageOptions.java:55)
	at com.google.cloud.ServiceOptions.getRpc(ServiceOptions.java:564)
	at com.google.cloud.storage.StorageOptions.getStorageRpcV1(StorageOptions.java:168)
	at com.google.cloud.storage.StorageImpl.<init>(StorageImpl.java:117)
	at com.google.cloud.storage.StorageOptions$DefaultStorageFactory.create(StorageOptions.java:51)
	at com.google.cloud.storage.StorageOptions$DefaultStorageFactory.create(StorageOptions.java:45)
	at com.google.cloud.ServiceOptions.getService(ServiceOptions.java:544)
	at org.wfanet.measurement.gcloud.gcs.GcsFromFlags.getStorage(GcsFromFlags.kt:29)
	at org.wfanet.measurement.gcloud.gcs.GcsStorageClient$Companion.fromFlags(GcsStorageClient.kt:103)
	at org.wfanet.measurement.duchy.deploy.gcloud.server.GcsSpannerComputationsServer$run$1$1.invokeSuspend(GcsSpannerComputationsServer.kt:52)
	at org.wfanet.measurement.duchy.deploy.gcloud.server.GcsSpannerComputationsServer$run$1$1.invoke(GcsSpannerComputationsServer.kt)
	at org.wfanet.measurement.duchy.deploy.gcloud.server.GcsSpannerComputationsServer$run$1$1.invoke(GcsSpannerComputationsServer.kt)
	at org.wfanet.measurement.gcloud.spanner.SpannerDatabaseConnector.usingSpanner(SpannerDatabaseConnector.kt:102)
	at org.wfanet.measurement.gcloud.spanner.SpannerDatabaseConnector$usingSpanner$1.invokeSuspend(SpannerDatabaseConnector.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.internal.ScopeCoroutine.afterResume(Scopes.kt:32)
	at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:102)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
	at kotlinx.coroutines.internal.ScopeCoroutine.afterResume(Scopes.kt:32)
	at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:102)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:280)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at org.wfanet.measurement.duchy.deploy.gcloud.server.GcsSpannerComputationsServer.run(GcsSpannerComputationsServer.kt:49)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1919)
	at picocli.CommandLine.access$1100(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2332)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2326)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2291)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2159)
	at picocli.CommandLine.execute(CommandLine.java:2058)
	at org.wfanet.measurement.common.CommandLinesKt.commandLineMain(CommandLines.kt:47)
	at org.wfanet.measurement.common.CommandLinesKt.commandLineMain$default(CommandLines.kt:42)
	at org.wfanet.measurement.duchy.deploy.gcloud.server.GcsSpannerComputationsServerKt.main(GcsSpannerComputationsServer.kt:71)
Caused by: java.lang.IllegalStateException: You are currently running with version 2.2.0 of google-api-client. You need at least version 1.31.1 of google-api-client to run version 1.32.1 of the Cloud Storage JSON API library.
	at com.google.common.base.Preconditions.checkState(Preconditions.java:534)
	at com.google.api.client.util.Preconditions.checkState(Preconditions.java:113)
	at com.google.api.services.storage.Storage.<clinit>(Storage.java:44)
	... 42 more
```

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -116,6 +116,8 @@ MAVEN_ARTIFACTS_DICT = dict(common_jvm_maven_artifacts_dict().items() + {
     "software.amazon.awssdk:acmpca": AWS_JAVA_SDK_VERSION,
     "com.google.crypto.tink:tink-gcpkms": "1.9.0",
     "com.google.crypto.tink:tink-awskms": "1.9.1",
+    "com.google.apis:google-api-services-storage": "v1-rev20240105-2.0.0",
+    "com.google.apis:google-api-services-sqladmin": "v1-rev20240101-2.0.0",
 }.items())
 
 EXCLUDED_MAVEN_ARTIFACTS = [x for x in COMMON_JVM_EXCLUDED_ARTIFACTS if x != "org.slf4j:slf4j-log4j12"] + ["org.apache.beam:beam-sdks-java-io-kafka"]

--- a/build/versions.bzl
+++ b/build/versions.bzl
@@ -21,6 +21,6 @@
 # * https://cloud.google.com/dataflow/docs/support/sdk-version-support-status#apache-beam-2.x-sdks
 # * https://beam.apache.org/documentation/runners/flink/#flink-version-compatibility
 # * https://docs.aws.amazon.com/kinesisanalytics/latest/java/earlier.html
-APACHE_BEAM_VERSION = "2.38.0"
+APACHE_BEAM_VERSION = "2.40.0"
 
 K8S_CLIENT_VERSION = "16.0.0"

